content/js1/sprints/2/prep/index.md

```diff
@@ -8,4 +8,406 @@ backlog= 'Module-JS1'
+
+Firstly, we need a way to track the **line of code that is currently executing.** In the diagrams below, we use an arrow to indicate which line of code is currently being executed. Each line is an instruction to the computer that will produce some effect. For example, a line of code with a variable declaration means "store a new variable with a particular value".
+
+Let's consider an example:
+
+<iframe width="800" height="300" frameborder="0" src="https://pythontutor.com/iframe-embed.html#code=const%20name%20%3D%20'Gayle'%3B&codeDivHeight=400&codeDivWidth=350&cumulative=false&curInstr=0&heapPrimitives=nevernest&origin=opt-frontend.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false"> </iframe>
+
+In the example above, we have a single variable declaration:
+
+```js
+const name = "Gayle";
+```
+
+which means store a string `"Gayle"` in memory.
+
+### 🖼️ Storing things in memory
+
+As variables are declared, we need a way to keep track of the memory that is stored as our program executes.
+In our diagrams, we keep track of memory using a _frame_.
+
+For now, we can think of a frame as a space in our diagram that represents the variables stored in memory in some part of our code.
+
+The _global frame_ is always the first frame that gets created when our program starts executing. We can think of the _global frame_ as a location for variables created outside of any function. When we run the variable declaration above, we get a string `"Gayle"` stored in the Global frame:
+
+![global-frame](global-frame.png)
+
+### 🏃 Executing functions
+
+So far we've seen 2 things:
+
+- lines of code are executed from top to bottom
+- we keep track of variables stored in memory using a global frame
+
+However, we need to consider what happens when we declare and call a function in our program.
+Let's consider another example:
+
+<iframe width="850" height="375" frameborder="0" src="https://pythontutor.com/iframe-embed.html#code=const%20firstName%20%3D%20'German'%3B%0Aconst%20lastName%20%3D%20'Bencii'%3B%0A%0Afunction%20makeGreeting%28%29%20%7B%0A%20%20%20%20const%20greeting%20%3D%20%60Hello%20there%60%3B%20%0A%7D%0A%0AmakeGreeting%28%29%3B&codeDivHeight=400&codeDivWidth=350&cumulative=false&curInstr=5&heapPrimitives=nevernest&origin=opt-frontend.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false"> </iframe>
+
+The first thing in the program is the function declaration for `makeGreeting`.
+
+`makeGreeting` is stored in the Global frame as an **object**. We don't need to explore objects as this point - all we need to understand at this stage is that objects are a data type in JavaScript. We will return to objects in more depth later on in the course.
+
+On line 11 we have a function call:
+
+```js {linenos=table,hl_lines=["7"],linenostart=1}
+function makeGreeting() {
+  const greeting = "Hello there!";
+}
+const firstName = "German";
+const lastName = "Bencci";
+
+makeGreeting();
```

Comment:
Let's also add one line _after_ this to show what happens _after_ `makeGreeting` is called (i.e. that the program resumes after the call)

Reply:
What would you put on that line? A "dummy" log? I feel that would just be distracting?

Reply:
Going to leave as is for the time being, but I agree we want to emphasise the resumption of the program after the call.  Need to think of a better example really.